MapzenSDK/MZMapViewController.swift

```diff
@@ -1177,21 +1071,29 @@ open class MZMapViewController: UIViewController, LocationManagerDelegate {
 extension MZMapViewController : TGMapViewDelegate, TGRecognizerDelegate {
   
   //MARK : TGMapViewDelegate
-  
-  open func mapView(_ mapView: TGMapViewController, didLoadSceneAsync scene: String) {
-    // if we loaded a house style scene looks something like: file:///var/containers/Bundle/Application/FAFA232A-1190-40CB-9391-7C9F44B51076/ios-sdk.app/housestyles.bundle/bubble-wrap/bubble-wrap-style-more-labels.yaml
-    guard let pathComponents = URL.init(string: scene)?.pathComponents else {
-      onStyleLoadedClosure = nil
+
+  open func mapView(_ mapView: TGMapViewController, didLoadScene sceneID: Int32, withError sceneError: Error?) {
+
+    //We only want to call back on the latest scene load - so we gate here to make sure we only call back on the latest.
+    //TODO: For 2.0 we should pass the Error along in the callback block.
+    if sceneID != latestSceneId {
       return
     }
-    // if we have path components, grab the last two (ie. bubble-wrap & bubble-wrap-style-more-labels.yaml), strip ".yaml" and check for existence in styles map
-    let sceneStyle = (pathComponents[pathComponents.count-2] + "/" + pathComponents.last!).replacingOccurrences(of: ".yaml", with: "")
-    guard let style = styles[sceneStyle] else {
-      onStyleLoadedClosure = nil
-      return
+    restoreBuiltInMarkers()
+    guard let styleClosure = sceneLoadCallback else { return }
+    styleClosure(currentStyle)
+    sceneLoadCallback = nil
+  }
+
+  func restoreBuiltInMarkers() {
+    //Handle built in marker restoration
+    if shouldShowCurrentLocation && currentLocationGem != nil {
+      let lastPosition = currentLocationGem?.point
+      currentLocationGem = nil
+      _ = showCurrentLocation(true)
+      currentLocationGem?.point = lastPosition!
```

Comment:
Any reason not to `guard let` on line 1091 and avoid force unwrapping here?

Reply:
Two issues with a guard statement:
- Guards require an exit of current scope when they fail, so the rest of the function won't execute in the event a guard check fails. I conceived this function doing more than just restoring the current location gem, so I was trying to set a pattern to not just "exit". Obviously this is not actually occurring right now, but it should mean less refactoring in the future. 
- Brevity - we do the check in the `if` statement to make sure `currentLocationGem` is not nil, so it's relatively safe to force unwrap. `point` itself is a required property in the protocol declaration, so the optional state is inherited from the location gem. 

However, thinking on this a bit, I think I could swap this out to using a `if let` and make everyone happy. Lemme give that a whirl.